File: catkin_ws/src/atr_pkg/src/ma_node.cpp
#include <ros/ros.h>
#include <std_msgs/String.h>
#include <qq_msgs/Carry.h>

void chao_callback(qq_msgs::Carry msg)
{
    ROS_WARN(msg.grade.c_str());
    ROS_WARN("%d 星", msg.star);
    ROS_INFO(msg.data.c_str());
    //printf("\n");
}

void yao_callback(qq_msgs::Carry msg)
{
    ROS_WARN(msg.data.c_str());
    //printf("\n");
}

int main(int argc, char *argv[])
{
    setlocale(LC_ALL,"");
    ros::init(argc, argv, "ma_node");

    ros::NodeHandle nh;
    ros::Subscriber sub = nh.subscribe("kuai_shang_che", 10, chao_callback);
    ros::Subscriber sub_2 = nh.subscribe("ge_ge", 10, yao_callback);
    
    while(ros::ok())
    {
        ros::spinOnce();
    }

    return 0;
}

File: catkin_ws/src/atr_pkg/scripts/ma_node.py
#!/usr/bin/env python3
#coding=utf-8

import rospy
from std_msgs.msg import String
from qq_msgs.msg import Carry

def chao_callback(msg):
    rospy.logwarn(msg.grade)
    rospy.logwarn(str(msg.star) + "星")
    rospy.loginfo(msg.data)

def yao_callback(msg):
    rospy.logwarn(msg.data)

if __name__ == "__main__":
    rospy.init_node("ma_node")

    sub = rospy.Subscriber("kuai_shang_che", Carry, chao_callback, queue_size=10)
    sub_2 = rospy.Subscriber("ge_ge", Carry, yao_callback, queue_size=10)

    rospy.spin()